### ISSUE TITLE
Incorrect paste behavior for bulleted lists

### ISSUE THREAD

**Issue by guitardave24** [facebook/draft-js#302]
I think there's an issue handling `<ul>` pastes.

Take a list that looks like this:
```
* one
* two
```

If you select all text with the mouse cursor, and then paste it. 

The result looks like:
```
one
*two
```

The HTML given to the paste processor above starts with: `<meta charset='utf-8'><ul ...`

However, if you select via cursor selection and make sure to select the *entire* block and focus on the end of the last block (which may need an extra left cursor movement) the result looks like this:
```
* one
* two
```
The HTML given to the paste processor in that scenario starts with: `<meta charset='utf-8'><li ...`

This is what I would have expected to occur in the first scenario. 

I looked into the paste processor and it seems to be ignoring `<ul>` elements and only looking for `<li>`. See https://github.com/facebook/draft-js/blob/master/src/model/encoding/convertFromHTMLToContentBlocks.js#L304

The ul is processed as an unstyled block. Perhaps just adding `ul` to getBlockTypeForTag is the fix?

Happy to submit a PR if this is agreed.

FWIW, I noticed this behavior in Facebook Notes too, but Medium does what I expect for sake of argument.

**Comment by hellendag** [comment]
cc @CLowbrow 

Sure seems like a bug that the first block would be treated as unstyled. If you want to put together a PR, I'd be open to it. Thanks!

**Comment by johanneslumpe** [comment]
For what it's worth: this bug applies to any kind of block, not just bulleted lists. If I have any two blocks and I copy them, then remove all content and paste them again, the first one will be pasted as `unstyled` block.

**Comment by johanneslumpe** [comment]
@hellendag Here is a gif of a fix I've been working on.
It properly pastes all block types and also retains the type of the fragment block.

![replace_type](https://cloud.githubusercontent.com/assets/3470207/14936763/61d955b2-0efd-11e6-90a8-b6d7e76f976b.gif)


For now I took the liberty to retain type of the block we are pasting into, when the block is not empty. If it is empty, the type will become the type of the pasted fragment. We can see though that this opens up some interesting edge cases, like when we are pasting new blocks the middle of a block.

I've just compared this with LibreOffice and here is an example of what happens there:

![libreoffice](https://cloud.githubusercontent.com/assets/3470207/14936775/9e71ce50-0efd-11e6-841f-61228d8db637.gif)

So my fix is not yet fully consistent with that behavior, but the question is: do we want it to be? For me the  LibreOffice behavior of removing the last list item when pasting at the end of a paragraph seems broken for example.
